llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -1352,9 +1352,10 @@ static bool runImpl(Module &M, AnalysisGetter &AG, TargetMachine &TM,
         return !AMDGPU::isEntryFunctionCC(Callee.getCallingConv()) &&
                (NumAssumedCallees <= IndirectCallSpecializationThreshold);
       };
-  AC.IPOAmendableCB = [](const Function &F) {
+  auto IPOAmendableCB = [](const Function &F) {
     return F.getCallingConv() == CallingConv::AMDGPU_KERNEL;
   };
+  AC.IPOAmendableCB = IPOAmendableCB;
```

Comment:
The fix looks good to me, but I wonder if the code authors might want to change these APIs as it looks really error prone to use.